### DISCUSSION TITLE
JsonDecodingException: Unexpected JSON token

### DISCUSSION THREAD

**Discussion by neronguyen** [supabase-community/supabase-kt#251] [Q&A] [answered]
I see the default Json contains ignoreUnknownKeys = true but why my app still throw this exception. I already try to install ContentNegotion with the custom Json with httpConfig {} but it doesn't work neither.

**Comment by jan-tennert** [comment]
never update the http config unless you know what you are doing, you could break the supabase client. What were you trying to do when this error occured?

**Reply by neronguyen** [reply]
I wanna config the Json factory like this:
```
        Json { 
            ignoreUnknownKeys = true
            coerceInputValues = true
            isLenient = true
            prettyPrint = true
        }
```

**Reply by neronguyen** [reply]
I just want to ignore that error with my json factory config

**Comment by neronguyen** [comment]
these config should take effect but not:
```
    @Provides
    @Singleton
    fun provideSupabaseClient(json: Json): SupabaseClient = createSupabaseClient(
        supabaseUrl = BuildConfig.SupabaseUrl,
        supabaseKey = BuildConfig.SupabaseAnonKey
    ) {
        install(GoTrue)
        install(Postgrest)
        install(Realtime)
        httpConfig {
            install(ContentNegotiation) {
                json(json)
            }
        }
    }
```

**Comment by jan-tennert** [comment, marked as answer]
I still don't know what you were doing when this error occured, but I think that should fix it:
```kotlin
defaultSerializer = KotlinXSerializer(json)
install(GoTrue)
install(Postgrest)
install(Realtime)
//no httpConfig!
```
